### ISSUE TITLE
Output the final url rather than the original url passed to the command

### ISSUE THREAD

**Issue by GlynnPhillips** [pa11y/pa11y#164]
#162 has highlighted the desire for the final (tested) url to be output rather than the original url that was passed to the command line.

I imagine this change wouldn't need to be a special case and that `pa11y` could always output the final url. Even in cases where the page is not being alter via the `beforeScript` option the page could still be redirected and it would be desirable to output the url that was actually tested.

**Comment by rowanmanning** [comment]
I like that idea. Potentially the report would be better if we indicated that the start URL and end URL are different? E.g.

```
Welcome to Pa11y

 > PhantomJS browser created
 > Testing the page "http://www.github.com"

Results for "https://github.com/" (redirected from "http://www.github.com/"):
...
```

**Comment by hollsk** [comment]
Should this issue be moved to [Truffler](https://github.com/springernature/truffler)?

**Comment by rowanmanning** [comment]
Yes I think it probably should maybe I'm not sure I'll check properly in a bit? 😂

**Comment by hollsk** [comment]
It shouldn't. Been having a look today and it's a maze of twisty passages, all alike. 

The `Results for "http://blahblah` line comes directly from the input itself (either entered in the CLI or in a config/task runner). The reporters don't really know anything about the final URL, and there's no way that I can see to get them to know about it without making the return results dirty. 

Somebody (think it was you, Rowan) added a debug log to the `testPage` function, that outputs the title of the current page ( https://github.com/pa11y/pa11y/blob/master/lib/pa11y.js ). If I pass the input URL into that function, I can get the current URL to compare it, and log an info notice if they don't match. 

Then we can remove the URL from the reporter and use a generic "Results from your test" statement instead. But we could only do it for the CLI reporter, because the HTML reports refer to the original URL in the header, and they wouldn't be too useful with no information. 

Or I could do the comparison in `testPage` and if a redirect happens, log a debug/info message instead and leave the CLI reporter alone entirely. 

Or there might be something I haven't thought of that would let us get the final URL into the reporter  without muddying the return results :/

**Comment by rowanmanning** [comment]
Unfortunately I think you're right, we can't do this without the results being muddied. I guess we could try sending a third parameter to the [`done` call here](https://github.com/pa11y/pa11y/blob/f51182de3f76e46d6b260d7c7ac70bf5607b6142/lib/pa11y.js#L82) but I'm not convinced that would work.

We might have to add the `breaking` label to this issue and wait for Pa11y v5 😔

**Comment by hollsk** [comment]
😭 

Shall I add the info notice at least? Somethin like

```
Welcome to Pa11y

 > PhantomJS browser created
 > Testing the page "http://www.nature.com/bonekey/archive/index.html"
 > Page was redirected at least once. Actual URL tested: "http://www.nature.com/bonekey/index.html"

Results for www.nature.com/bonekey/archive/index.html:
```

**Comment by rowanmanning** [comment]
This will be released as part of Pa11y 5.0
